src/controller/CHIPDeviceController.cpp

```diff
@@ -723,6 +723,8 @@ CHIP_ERROR DeviceCommissioner::UnpairDevice(NodeId remoteDeviceId)
 {
     // TODO: Send unpairing message to the remote device.
 
+    FreeRendezvousSession();
```

Comment:
Need to check `remoteDeviceId`?

Reply:
You means this check "VerifyOrReturnError(device->GetDeviceId() == remoteDeviceId, CHIP_ERROR_INVALID_DEVICE_DESCRIPTOR);"

I think this check might not apply to UnpairDevice since it might call on the device which is not currently being paired.

Reply:
Right, why would unpairing a different device free the rendezvous session?

Reply:
That make sense, I will take a look